### ISSUE TITLE
Nvidia-docker2 rpms unavailable with older docker versions for centos7

### ISSUE THREAD

**Issue by aashish-sheshadri** [NVIDIA/nvidia-docker#896]
_The template below is mostly useful for bug reports and support questions. Feel free to remove anything which doesn't apply to you and add more information where it makes sense._

_Also, before reporting a new issue, please make sure that:_

- _You read carefully the [documentation and frequently asked questions](https://github.com/NVIDIA/nvidia-docker/wiki)._
- _You [searched](https://github.com/NVIDIA/nvidia-docker/issues?utf8=%E2%9C%93&q=is%3Aissue) for a similar issue and this is not a duplicate of an existing one._
- _This issue is not related to [NGC](https://github.com/NVIDIA/nvidia-docker/wiki/NGC), otherwise, please use the [devtalk forums](https://devtalk.nvidia.com/default/board/200/nvidia-gpu-cloud-ngc-users/) instead._
- _You went through the [troubleshooting](https://github.com/NVIDIA/nvidia-docker/wiki/Troubleshooting) steps._

---

### 1. Issue or feature description
Unable to locate Nvidia-docker2 for Docker 18.06.0-ce
### 3. Information to [attach](https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/) (optional if deemed irrelevant)

 - [ ] Some nvidia-container information: `nvidia-container-cli -k -d /dev/tty info`
 - [x] Kernel version from `uname -a`
 - [ ] Any relevant kernel output lines from `dmesg`
 - [ ] Driver information from `nvidia-smi -a`
 - [x] Docker version from `docker version`
 - [ ] NVIDIA packages version from `dpkg -l '*nvidia*'` _or_ `rpm -qa '*nvidia*'`
 - [ ] NVIDIA container library version from `nvidia-container-cli -V`
 - [ ] NVIDIA container library logs (see [troubleshooting](https://github.com/NVIDIA/nvidia-docker/wiki/Troubleshooting))
 - [ ] Docker command, image and tag used

**Comment by aashish-sheshadri** [comment]
I used the instructions here listed under: CentOS 7 (docker-ce), RHEL 7.4/7.5 (docker-ce), Amazon Linux 1/2

yum --disablerepo="*" --enablerepo="nvidia-container-runtime" list available
Available Packages
nvidia-container-runtime.x86_64    2.0.0-1.docker18.09.0    nvidia-container-runtime

yum --disablerepo="*" --enablerepo="nvidia-docker" list available
Available Packages
nvidia-docker.x86_64    1.0.1-1     nvidia-docker
nvidia-docker2.noarch   2.0.3-1.docker18.09.0.ce   nvidia-docker

So I only see packages built for 18.09.

**Comment by coreyryanhanson** [comment]
It's weird, I'm not seeing that no matter what I do. Could the GPG key check failing be redirecting me to an outdated mirror? I've been getting that message for a bit, and in the past, tried updating the repo keys per this instruction: https://nvidia.github.io/nvidia-docker/ but it did not work. This was the output from just now when I tried to install fresh after deleting the repos:
``
user@localhost:~$ docker --version
Docker version 18.09.0, build 4d60db4
user@localhost:~$ distribution=$(. /etc/os-release;echo $ID$VERSION_ID)
user@localhost:~$ curl -s -L https://nvidia.github.io/nvidia-docker/$distribution/nvidia-docker.repo | sudo tee /etc/yum.repos.d/nvidia-docker.repo
[sudo] password for user: 
[libnvidia-container]
name=libnvidia-container
baseurl=https://nvidia.github.io/libnvidia-container/centos7/$basearch
repo_gpgcheck=1
gpgcheck=0
enabled=1
gpgkey=https://nvidia.github.io/libnvidia-container/gpgkey
sslverify=1
sslcacert=/etc/pki/tls/certs/ca-bundle.crt

[nvidia-container-runtime]
name=nvidia-container-runtime
baseurl=https://nvidia.github.io/nvidia-container-runtime/centos7/$basearch
repo_gpgcheck=1
gpgcheck=0
enabled=1
gpgkey=https://nvidia.github.io/nvidia-container-runtime/gpgkey
sslverify=1
sslcacert=/etc/pki/tls/certs/ca-bundle.crt

[nvidia-docker]
name=nvidia-docker
baseurl=https://nvidia.github.io/nvidia-docker/centos7/$basearch
repo_gpgcheck=1
gpgcheck=0
enabled=1
gpgkey=https://nvidia.github.io/nvidia-docker/gpgkey
sslverify=1
sslcacert=/etc/pki/tls/certs/ca-bundle.crt
user@localhost:~$ sudo yum install -y nvidia-docker2
Loaded plugins: fastestmirror, langpacks, nvidia
Loading mirror speeds from cached hostfile
 * base: mirror.cc.columbia.edu
 * elrepo: reflector.westga.edu
 * epel: mirror.math.princeton.edu
 * extras: mirror.cc.columbia.edu
 * nux-dextop: mirror.li.nux.ro
 * rpmfusion-free-updates: mirror.math.princeton.edu
 * rpmfusion-nonfree-updates: mirror.math.princeton.edu
 * updates: mirror.cc.columbia.edu
libnvidia-container/x86_64/signature                     |  455 B     00:00     
Retrieving key from https://nvidia.github.io/libnvidia-container/gpgkey
libnvidia-container/x86_64/signature                     | 2.0 kB     00:00 !!! 
https://nvidia.github.io/libnvidia-container/centos7/x86_64/repodata/repomd.xml: [Errno -1] repomd.xml signature could not be verified for libnvidia-container
Trying other mirror.
nvidia-container-runtime/x86_64/signature                |  455 B     00:00     
Retrieving key from https://nvidia.github.io/nvidia-container-runtime/gpgkey
nvidia-container-runtime/x86_64/signature                | 2.0 kB     00:00 !!! 
https://nvidia.github.io/nvidia-container-runtime/centos7/x86_64/repodata/repomd.xml: [Errno -1] repomd.xml signature could not be verified for nvidia-container-runtime
Trying other mirror.
nvidia-docker/x86_64/signature                           |  455 B     00:00     
Retrieving key from https://nvidia.github.io/nvidia-docker/gpgkey
nvidia-docker/x86_64/signature                           | 2.0 kB     00:00 !!! 
https://nvidia.github.io/nvidia-docker/centos7/x86_64/repodata/repomd.xml: [Errno -1] repomd.xml signature could not be verified for nvidia-docker
Trying other mirror.
Resolving Dependencies
--> Running transaction check
---> Package nvidia-docker2.noarch 0:2.0.3-1.docker18.03.1.ce will be installed
--> Processing Dependency: docker-ce = 18.03.1.ce for package: nvidia-docker2-2.0.3-1.docker18.03.1.ce.noarch
--> Finished Dependency Resolution
--> Finding unneeded leftover dependencies
Found and removing 0 unneeded dependencies
Error: Package: nvidia-docker2-2.0.3-1.docker18.03.1.ce.noarch (nvidia-docker)
           Requires: docker-ce = 18.03.1.ce
           Installed: 3:docker-ce-18.09.0-3.el7.x86_64 (@docker-ce-stable)
               docker-ce = 3:18.09.0-3.el7
           Available: docker-ce-17.03.0.ce-1.el7.centos.x86_64 (docker-ce-stable)
               docker-ce = 17.03.0.ce-1.el7.centos
           Available: docker-ce-17.03.1.ce-1.el7.centos.x86_64 (docker-ce-stable)
               docker-ce = 17.03.1.ce-1.el7.centos
           Available: docker-ce-17.03.2.ce-1.el7.centos.x86_64 (docker-ce-stable)
               docker-ce = 17.03.2.ce-1.el7.centos
           Available: docker-ce-17.03.3.ce-1.el7.x86_64 (docker-ce-stable)
               docker-ce = 17.03.3.ce-1.el7
           Available: docker-ce-17.06.0.ce-1.el7.centos.x86_64 (docker-ce-stable)
               docker-ce = 17.06.0.ce-1.el7.centos
           Available: docker-ce-17.06.1.ce-1.el7.centos.x86_64 (docker-ce-stable)
               docker-ce = 17.06.1.ce-1.el7.centos
           Available: docker-ce-17.06.2.ce-1.el7.centos.x86_64 (docker-ce-stable)
               docker-ce = 17.06.2.ce-1.el7.centos
           Available: docker-ce-17.09.0.ce-1.el7.centos.x86_64 (docker-ce-stable)
               docker-ce = 17.09.0.ce-1.el7.centos
           Available: docker-ce-17.09.1.ce-1.el7.centos.x86_64 (docker-ce-stable)
               docker-ce = 17.09.1.ce-1.el7.centos
           Available: docker-ce-17.12.0.ce-1.el7.centos.x86_64 (docker-ce-stable)
               docker-ce = 17.12.0.ce-1.el7.centos
           Available: docker-ce-17.12.1.ce-1.el7.centos.x86_64 (docker-ce-stable)
               docker-ce = 17.12.1.ce-1.el7.centos
           Available: docker-ce-18.03.0.ce-1.el7.centos.x86_64 (docker-ce-stable)
               docker-ce = 18.03.0.ce-1.el7.centos
           Available: docker-ce-18.03.1.ce-1.el7.centos.x86_64 (docker-ce-stable)
               docker-ce = 18.03.1.ce-1.el7.centos
           Available: docker-ce-18.06.0.ce-3.el7.x86_64 (docker-ce-stable)
               docker-ce = 18.06.0.ce-3.el7
           Available: docker-ce-18.06.1.ce-3.el7.x86_64 (docker-ce-stable)
               docker-ce = 18.06.1.ce-3.el7
 You could try using --skip-broken to work around the problem
You could try running: rpm -Va --nofiles --nodigest

**Comment by RenaudWasTaken** [comment]
@aashish-sheshadri your yum command is incorrect:
```
yum --disablerepo="*" --enablerepo="nvidia-docker" --showduplicates list available
Loaded plugins: fastestmirror, ovl
Loading mirror speeds from cached hostfile
Available Packages
nvidia-docker.x86_64                                                                              1.0.0-1                                                                                              nvidia-docker
nvidia-docker.x86_64                                                                              1.0.1-1                                                                                              nvidia-docker
nvidia-docker2.noarch                                                                             2.0.0-1.docker1.12.6                                                                                 nvidia-docker
nvidia-docker2.noarch                                                                             2.0.0-1.docker17.03.2.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.0-1.docker17.06.1.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.0-1.docker17.06.2.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.0-1.docker17.09.0.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.1-1.docker1.12.6                                                                                 nvidia-docker
nvidia-docker2.noarch                                                                             2.0.1-1.docker1.13.1                                                                                 nvidia-docker
nvidia-docker2.noarch                                                                             2.0.1-1.docker17.03.2.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.1-1.docker17.06.2.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.1-1.docker17.09.0.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.1-1.docker17.09.1.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.2-1.docker1.12.6                                                                                 nvidia-docker
nvidia-docker2.noarch                                                                             2.0.2-1.docker1.13.1                                                                                 nvidia-docker
nvidia-docker2.noarch                                                                             2.0.2-1.docker17.03.2.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.2-1.docker17.06.2.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.2-1.docker17.09.0.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.2-1.docker17.09.1.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.2-1.docker17.12.0.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.3-1.docker1.12.6                                                                                 nvidia-docker
nvidia-docker2.noarch                                                                             2.0.3-1.docker1.13.1                                                                                 nvidia-docker
nvidia-docker2.noarch                                                                             2.0.3-1.docker17.03.2.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.3-1.docker17.06.2.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.3-1.docker17.09.0.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.3-1.docker17.09.1.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.3-1.docker17.12.0.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.3-1.docker17.12.1.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.3-1.docker18.03.0.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.3-1.docker18.03.1.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.3-1.docker18.06.0.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.3-1.docker18.06.1.ce                                                                             nvidia-docker
nvidia-docker2.noarch                                                                             2.0.3-1.docker18.09.0.ce                                                                             nvidia-docker
```
